Scripts/Proxies/TwoHandedProxyBase.cs

```diff
@@ -277,14 +338,84 @@ void RemoveFeedbackRequest(ProxyFeedbackRequest request)
 
         public void ClearFeedbackRequests(IRequestFeedback caller)
         {
-            var requests = caller == null
-                ? new List<ProxyFeedbackRequest>(m_FeedbackRequests)
-                : m_FeedbackRequests.Where(feedbackRequest => feedbackRequest.caller == caller).ToList();
+            // Interate over keys instead of pairs in the dictionary, in order to prevent out-of-sync errors when exiting EXR
+            foreach (var key in m_FeedbackRequests.Keys.ToList())
+            {
+                if (key != null && key.caller == caller)
+                    RemoveFeedbackRequest(key);
+            }
+        }
+
+        void UpdateVisibility()
+        {
+            var rightProxyRequestsExist = false;
+            var leftProxyRequestsExist = false;
+            var shakenVisibility = m_SemitransparentLockRequest != null;
+            if (shakenVisibility)
+            {
+                // Left & right device affordances should be visible when the input-device is shaken
+                rightProxyRequestsExist = true;
+                leftProxyRequestsExist = true;
+            }
+            else if (m_FeedbackRequests.Count > 0)
+            {
+                // Find any visible feedback requests for each hand
+                foreach (var request in m_FeedbackRequests)
+                {
+                    var node = request.Key.node;
+                    var visible = request.Key.visible;
+
+                    if (!leftProxyRequestsExist)
+                        leftProxyRequestsExist = node == Node.LeftHand && visible;
+
+                    if (!rightProxyRequestsExist)
+                        rightProxyRequestsExist = node == Node.RightHand && visible;
+
+                    if (rightProxyRequestsExist && leftProxyRequestsExist)
+                        break;
+                }
+            }
+
+            rightAffordanceRenderersVisible = rightProxyRequestsExist;
+            leftAffordanceRenderersVisible = leftProxyRequestsExist;
+
+            rightBodyRenderersVisible = shakenVisibility;
+            leftBodyRenderersVisible = shakenVisibility;
+        }
+
+        IEnumerator MonitorFeedbackRequestLifespan(ProxyFeedbackRequest request)
+        {
+            if (request.proxyShaken)
+            {
+                if (m_SemitransparentLockRequest != null)
+                    yield break;
 
-            foreach (var feedbackRequest in requests)
+                m_SemitransparentLockRequest = request;
+            }
+
+            request.visible = true;
+
+            const float kShakenVisibilityDuration = 5f;
+            const float kShorterOpaqueDurationScalar = 0.125f;
+            float duration = request.proxyShaken ? kShakenVisibilityDuration : k_DefaultFeedbackDuration * kShorterOpaqueDurationScalar;
+            var currentDuration = 0f;
+            while (request != null && currentDuration < duration)
```

Comment:
`request` is always non-null, and you're accessing its properties earlier here anyway. Could just start this coroutine with `if (request == null) yield break;`, but that shouldn't be necessary.